test/jb_controller/reconfigure_funding_cycles_of.test.js

```diff
@@ -0,0 +1,537 @@
+        ),
+    ).to.equal(timestamp);
+
+    let tx = jbController
+      .connect(caller)
+      .reconfigureFundingCyclesOf(
+        PROJECT_ID,
+        fundingCycleData,
+        fundingCycleMetadata.unpacked,
+        groupedSplits,
+        fundAccessConstraints,
+      );
+
+    await Promise.all(
+      fundAccessConstraints.map(async (constraints) => {
+        await expect(tx)
+          .to.emit(jbController, 'SetFundAccessConstraints')
+          .withArgs(
+            timestamp,
+            1,
+            PROJECT_ID,
+            [
+              constraints.terminal,
+              constraints.distributionLimit,
+              constraints.overflowAllowance,
+              constraints.currency,
+            ],
+            caller.address,
+          );
+      }),
+    );
+  });
+
+  it(`Can't reconfigure funding cycle if caller is not authorized`, async function () {
+    const {
+      jbController,
+      projectOwner,
+      addrs,
+      fundingCycleData,
+      fundingCycleMetadata,
+      splits,
+      mockJbOperatorStore,
+      mockTerminal1,
+      mockTerminal2,
+    } = await setup();
+
+    const caller = addrs[0];
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(caller.address, projectOwner.address, PROJECT_ID, RECONFIGURE_INDEX)
+      .returns(false);
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(caller.address, projectOwner.address, 0, RECONFIGURE_INDEX)
+      .returns(false);
+
+    const groupedSplits = [{ group: 1, splits }];
+    const terminals = [mockTerminal1.address, mockTerminal2.address];
+    const fundAccessConstraints = makeFundingAccessConstraints({ terminals });
+
+    let tx = jbController
+      .connect(caller)
+      .reconfigureFundingCyclesOf(
+        PROJECT_ID,
+        fundingCycleData,
+        fundingCycleMetadata.unpacked,
+        groupedSplits,
+        fundAccessConstraints,
+      );
+
+    await expect(tx).to.be.revertedWith(errors.UNAUTHORIZED);
+  });
+
+  it(`Should reconfigure funding cycle without grouped splits`, async function () {
+    const {
+      jbController,
+      projectOwner,
+      timestamp,
+      fundingCycleData,
+      fundingCycleMetadata,
+      splits,
+      mockTerminal1,
+      mockTerminal2,
+    } = await setup();
+
+    const terminals = [mockTerminal1.address, mockTerminal2.address];
+    const fundAccessConstraints = makeFundingAccessConstraints({ terminals });
+
+    expect(
+      await jbController
+        .connect(projectOwner)
+        .callStatic.reconfigureFundingCyclesOf(
+          PROJECT_ID,
+          fundingCycleData,
+          fundingCycleMetadata.unpacked,
+          [],
+          fundAccessConstraints,
+        ),
+    ).to.equal(timestamp);
+
+    let tx = jbController
+      .connect(projectOwner)
+      .reconfigureFundingCyclesOf(
+        PROJECT_ID,
+        fundingCycleData,
+        fundingCycleMetadata.unpacked,
+        [],
+        fundAccessConstraints,
+      );
+
```

Comment:
curious if during these tests we should should also be checking the set `distributionLimitOf`, `overflowAllowanceOf`, and `currencyOf`?

Reply:
Good catch, I got lost between PR merged, the gas optim is still open, right? I'll add these tests when main-merged, so this is a review of "final-but-not-so-final-draft-v2-copy" version,:)

Reply:
perfect!